assets/chat/js/formatters/UrlFormatter.js

```diff
@@ -71,14 +72,20 @@ export default class UrlFormatter {
       const m = decodedUrl.match(self.linkregex);
       if (m) {
         const encodedUrl = self.encodeUrl(m[0]);
+        const embedHashLink = this.hashLinkConverter.convert(
+          encodedUrl,
+          /* throwErrors */ false
+        );
         const maxUrlLength = 90;
         let urlText = encodedUrl;
         if (urlText.length > maxUrlLength) {
           urlText = `${urlText.slice(0, 40)}...${urlText.slice(-40)}`;
         }
         const extra = self.encodeUrl(decodedUrl.substring(m[0].length));
         const href = `${scheme ? '' : 'http://'}${encodedUrl}`;
-        return `<a target="_blank" class="externallink ${extraclass}" href="${href}" rel="nofollow">${urlText}</a>${extra}`;
+        return embedHashLink
+          ? `<a target="_blank" class="externallink ${extraclass}" href="${href}" rel="nofollow">${urlText}</a><button class="embed-button" onclick="location.href='/bigscreen${embedHashLink}'">Embed</button>`
+          : `<a target="_blank" class="externallink ${extraclass}" href="${href}" rel="nofollow">${urlText}</a>${extra}`;
```

Comment:
Would you mind using an anchor over a button? That would make the click handler unnecessary.

Reply:
Yep, no worries